### DISCUSSION TITLE
Unable to create producer

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#22405] [Q&A and General discussion]
When I insert data, usually after 20 million or more, I get an error that I can't create a producer. I usually clean the pulsar message queue, that is, clear the backlog, but then the inserted data will probably not be used, it won't load, Rebuilding the index doesn't help either。
**What is the right thing to do in this situation?**
#########################################################

That's what I usually do
```
bin/pulsar-admin topics stats public/default/by-dev-rootcoord-dml_10  # check the backlog size and subscription name
bin/pulsar-admin topics clear-backlog public/default/by-dev-rootcoord-dml_10 -s <subscription-name> 
```
I don't think that's a good idea

When the insertion concurrency is too large,rootcoord Failed to create producer

**Comment by yhmo** [comment]
Pulsar is like a WAL component for Milvus. All the insert requests will be firstly stored into Pulsar before processed by data node. The pulsar could be insufficient storage after large amount of insert requests. So, first check the free space of the pulsar server.

**Reply by ingale726** [reply]
My guess is that a lot of insert requests go through Pulsar, only a few are inserted, and the rest is stored in Pulsar, so the Pulsar stores more and more, eventually reaching its maximum capacity, mainly because the insert speed is too slow. The hard drive isn't full of storage, but the Pulsar is full

**Reply by yhmo** [reply]
> My guess is that a lot of insert requests go through Pulsar, only a few are inserted, and the rest is stored in Pulsar, so the Pulsar stores more and more, eventually reaching its maximum capacity, mainly because the insert speed is too slow. The hard drive isn't full of storage, but the Pulsar is full

Exactly. In the milvus.yaml, there is a configuration:
```
common:
    retentionDuration: 86400 # time travel reserved time, insert/delete will not be cleaned in this period. 1 days in seconds
```

By default, the insert/delete requests will be kept in pulsar/kafka in 1 day. And pulsa/kafa is an independent service, it also has replica machinery for cluster mode. So, if you inserted 100Gb data within 1 day, you need to assign 100*x Gb memory for the pulsar/kafka server.
